core/src/main/scala/org/apache/spark/deploy/SparkHadoopUtil.scala

```diff
@@ -201,6 +199,30 @@ class SparkHadoopUtil extends Logging {
     val baseStatus = fs.getFileStatus(basePath)
     if (baseStatus.isDir) recurse(basePath) else Array(baseStatus)
   }
+
+  private val varPattern = """(\$\{hadoop-[^\}\$\u0020]+\})""".r.unanchored
+
+  /**
+   * Substitute variables by looking them up in Hadoop configs. Only variables that match the
+   * ${hadoop- .. } pattern are substituted.
+   */
+  def substituteHadoopVariables(text: String, hadoopConf: Configuration): String = {
+    text match {
+      case varPattern(matched) => {
+        val key = matched.substring(9, matched.length() - 1) // remove ${hadoop- .. }
+        val eval = Option[String](hadoopConf.get(key))
+          .map { value => text.replace(matched, value) }
+        if (eval.isEmpty) {
+          // The variable was not found in Hadoop configs, so return text as is.
```

Comment:
it might be nice to put logDebug in here in case someone trying to debug the regex pattern